pkg/nrtupdater/nrtupdater.go

```diff
@@ -108,15 +110,28 @@ func (te *NRTUpdater) Update(info MonitorInfo) error {
 	return nil
 }
 
-func (te *NRTUpdater) Run(infoChannel <-chan MonitorInfo) chan<- struct{} {
+func (te *NRTUpdater) NewCondition(status v1.ConditionStatus) v1.PodCondition {
```

Comment:
it seems you are never using `te` in the body of the method. So let's make this method a free function.